tests/model/model.cpp

```diff
@@ -552,3 +554,48 @@ TEST(Model, setAndCheckIdsAllEntities)
     const std::string a = printer->printModel(m);
     EXPECT_EQ(a, e);
 }
+
+TEST(Model, missingUnitsFromImportOfCnTerms)
```

Comment:
I feel like I'm missing the point of this test.  The `cn` term has a units defined in the model so why is it missing?  The missing here refers to something but I know it is not the clearly not missing units but it obviously is for something else which I can't figure out.  Anyone able to clarify what this is actually doing.  As far as I can tell the import is the same model?

Reply:
It might have been lost in the merge ... but the point of the test is to show that when we currently import a model which uses <cn> to define units, we haven't previously been importing those units too, just the ones associated with variables.

Reply:
So the test shouldn't already have the units defined then?  If they units we need for the `cn` are already defined then we don't need to import them?

Reply:
The test should import a model which uses <cn> units that are not also used by a variable.  That was the point of the whole PR - we were only importing units used by variables, and ignoring those used exclusively in the <cn> tags.

Reply:
Yes I agree but this test doesn't show me that if you import a component from another model that makes use of a `cn` element in some math then the units for that `cn` are imported as well.

Reply:
OK, I've found the confusion and updated the comments for others too.  Basically this uses two methods to create a model - one via the `Parser` of the string that's specified in the test, and the other by importing the component from a file which contains exactly the same string.   The `Parser` was creating a valid model, but the import was missing out on importing the units.  The test just demonstrates that this bug is now fixed :)